sdk/storage/AzureStorageBlob/Source/Transport/URLSessionTransferManager.swift

```diff
@@ -394,13 +394,57 @@ internal final class URLSessionTransferManager: NSObject, TransferManager, URLSe
                     blockTransfer.state = .pending
                 }
             }
+            reconnectClient(for: transfer)
             queueOperationsFor(blobTransfer: transfer)
         default:
             fatalError("Unrecognized transfer type: \(transfer.self)")
         }
         operation(transfer.operation, didChangeState: transfer.state)
     }
 
+    func reconnectClient(for transfer: BlobTransfer) {
+        // early out if a client is already connected
+        switch transfer.transferType {
+        case .upload:
+            guard transfer.uploader == nil else { return }
+        case .download:
+            guard transfer.downloader == nil else { return }
+        }
+
+        // must create one
+        guard let client = delegate?.client(forRestorationId: transfer.clientRestorationId) as? StorageBlobClient,
+            let options = delegate?.options(forRestorationId: transfer.clientRestorationId) else {
+            fatalError("Cannot recreate uploader/downloader!")
+        }
+        switch transfer.transferType {
+        case .upload:
+            guard let uploadOptions = options as? UploadBlobOptions else { return }
+            guard let sourceUrl = transfer.source else { return }
+            guard let destUrl = transfer.destination else { return }
+            // FIXME: Fix blobProperties!
+            let blobProperties = BlobProperties(contentType: "image/jpg")
```

Comment:
I really don't love the approach I took here, and this is part of the reason why. I really don't want another delegate `properties(forRestorationId:)`. Maybe these could be serialized in the model.

Would love to discuss this!

Reply:
Still need to discuss, so will leave this open. But not blocking merge for this PR.